packages/utils-debugging-protocol-common/src/debugging-protocol-connector.ts

```diff
@@ -22,26 +22,257 @@ import { Crdp } from 'chrome-remote-debug-protocol';
 import { createCDPAsyncHTMLDocument, CDPAsyncHTMLDocument, AsyncHTMLElement } from './cdp-async-html';
 import { getContentTypeData, getType } from 'hint/dist/src/lib/utils/content-type';
 import { debug as d } from 'hint/dist/src/lib/utils/debug';
-import * as logger from 'hint/dist/src/lib/utils/logging';
+// import * as logger from 'hint/dist/src/lib/utils/logging';
 import cutString from 'hint/dist/src/lib/utils/misc/cut-string';
 import delay from 'hint/dist/src/lib/utils/misc/delay';
-import hasAttributeWithValue from 'hint/dist/src/lib/utils/network/has-attribute-with-value';
 import isHTMLDocument from 'hint/dist/src/lib/utils/network/is-html-document';
 
 import {
     BrowserInfo, IConnector,
-    IAsyncHTMLElement, ElementFound, Event, FetchEnd, FetchError, ILauncher, TraverseUp, TraverseDown,
+    ElementFound, Event, FetchEnd, FetchError, ILauncher, TraverseUp, TraverseDown,
     Response, Request, NetworkData, HttpHeaders
 } from 'hint/dist/src/lib/types';
 
 import { normalizeHeaders } from '@hint/utils-connector-tools/dist/src/normalize-headers';
-import { RedirectManager } from '@hint/utils-connector-tools/dist/src/redirects';
 import { Requester } from '@hint/utils-connector-tools/dist/src/requester';
 
 import { Engine } from 'hint/dist/src/lib/engine';
+import { ContextReplacementPlugin } from 'webpack';
 
 const debug: debug.IDebugger = d(__filename);
 
+enum RequestStatus {
+    willBeSent = 'willBeSent',
+    responseReceived = 'responseReceived',
+    loadingFinished = 'loadingFinished',
+    loadingFailed = 'loadingFailed'
+}
+
+type NetworkEvent = Crdp.Network.RequestWillBeSentEvent |
+    Crdp.Network.ResponseReceivedEvent |
+    Crdp.Network.LoadingFinishedEvent |
+    Crdp.Network.LoadingFailedEvent;
+
+let overrideInvalidCert: false;
```

Comment:
IMO, this should be a private variable in each class, and it should be pass through the constructor.